### PR TITLE
Add player action signal

### PR DESCRIPTION
**Pull request by fourlastor**
This PR adds new signal `OnAction` to `Player`, which is fired whenever the user presses the `ui_select` action.

The facing direction of the player is updated in the animation player for the various idle/run animations

### CODE REVIEW

File: tilemap/Map.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using Godot;
+using System;
+
+public class Map : Node2D
+{
+    // Declare member variables here. Examples:
+    // private int a = 2;
```
Comment (takecare) [inline tilemap/Map.cs]: this can be removed 😄

File: player/Player.cs [hunk 2]
```diff
@@ -6,17 +6,34 @@ public class Player : KinematicBody2D
...
 
     public override void _Ready()
     {
         base._Ready();
-        animationTree = (AnimationTree)GetNode("AnimationTree");
+        animationTree = GetNode<AnimationTree>("AnimationTree");
         animationStateMachine = (AnimationNodeStateMachinePlayback)animationTree.Get("parameters/playback");
     }
 
+    public override void _UnhandledInput(InputEvent @event)
```
Comment (takecare) [inline player/Player.cs]: i reckon this should be `_UnhandledKeyInput` instead and (personal pref:) it should be moved to the bottom (after `_PhysicsProcess`)

File: player/Player.cs [hunk 3]
```diff
@@ -6,17 +6,34 @@ public class Player : KinematicBody2D
...
     private AnimationTree animationTree;
     private AnimationNodeStateMachinePlayback animationStateMachine;
     private bool isRunning = false;
+    private Vector2 facingDirection = Vector2.One;
 
     public override void _Ready()
     {
         base._Ready();
-        animationTree = (AnimationTree)GetNode("AnimationTree");
+        animationTree = GetNode<AnimationTree>("AnimationTree");
```
Comment (takecare) [inline player/Player.cs:20]: this is another downside of using c# 😄

File: player/Player.tscn [hunk 4]
```diff
@@ -39,6 +39,20 @@ tracks/0/keys = {
 "update": 1,
 "values": [ 0, 4, 8, 12 ]
 }
+tracks/1/type = "method"
```
Comment (takecare) [inline player/Player.tscn]: i don't think we should rely on the animation to keep track of the direction. i think it's better if we do it in the code as it avoids the signal and it's more visible. we can have something like the following in the `Player`'s `_PhysicsProcess`:

```csharp
var x = movement.x > 0 ? 1 : (movement.x < 0 ? -1 : 0);
var y = movement.y > 0 ? 1 : (movement.y < 0 ? -1 : 0);
GD.Print("direction=" + new Vector2(x, y));
```

what do you think?

File: tilemap/Map.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    private Player player;
+
+    // Called when the node enters the scene tree for the first time.
```
Comment (takecare) [inline tilemap/Map.cs]:
```suggestion
```

File: tilemap/Map.cs (hunk 1 above)
Comment (fourlastor) [reply tilemap/Map.cs]: [21bd372](https://github.com/fourlastor-jams/game-off-2022/pull/13/commits/21bd372b22ee0250ea70588a63e39dc9e7448937) :see_no_evil:

File: player/Player.cs (hunk 2 above)
Comment (fourlastor) [reply player/Player.cs]: [75f76af](https://github.com/fourlastor-jams/game-off-2022/pull/13/commits/75f76afb2d7d068e55c43102cd6e89ebfd2d1f10) TIL!

File: player/Player.tscn (hunk 4 above)
Comment (fourlastor) [reply player/Player.tscn]: [8002e72](https://github.com/fourlastor-jams/game-off-2022/pull/13/commits/8002e72bdeaa2200277be3f3f6b8266eeb09cd6c) makes sense, I used `Mathf.Sign` for the direction (it should be 1/-1 all the time as the character always faces a direction)

**Review by takecare — APPROVED** [review]

**Pull request merged**